substrate/frame/tips/src/lib.rs

```diff
@@ -336,10 +342,13 @@ pub mod pallet {
 			let tipper = ensure_signed(origin)?;
 			let who = T::Lookup::lookup(who)?;
 			ensure!(T::Tippers::contains(&tipper), BadOrigin);
+
 			let reason_hash = T::Hashing::hash(&reason[..]);
 			ensure!(!Reasons::<T, I>::contains_key(&reason_hash), Error::<T, I>::AlreadyKnown);
-			let hash = T::Hashing::hash_of(&(&reason_hash, &who));
 
+			ensure!(T::MaxTipAmount::get() >= tip_value, Error::<T, I>::MaxTipAmountExceeded);
```

Comment:
Feels like this should be done before you calculate the `reason_hash`, otherwise you'd be doing work only to be discarded later.

Reply:
Yes, I have thought about this before. However, I want to maintain the same business logic as before while making it more intuitive. Moreover, I can place this code before `T::Tippers::contains()` to avoid reading from council storage.

But if you believe it would be better to put this code before the `reason_hash`, I will make that change.

Reply:
Your comments feels like it was made from a place of fear. What kind of things do you feel you'd break when you make such a change?